src/components/Site/Logger.tsx

```diff
@@ -32,21 +36,58 @@ export function LoggerTab({
     logger,
     logger_uid,
 }: LoggerTabProps): ReactElement {
+    const channelTemplates = useSelector((state: RootState) => state.templates);
+
     const [infoExpanded, setInfoExpanded] = useState(false);
 
     const handleInfoButton = () => setInfoExpanded(!infoExpanded);
 
-    const columns: GridColDef[] = [
-        { field: 'timestamp', headerName: 'timestamp', flex: 1 },
+    const dateFormat = 'M/D/YYYY hh:mm:ss:SSS A';
+
+    const [
+        gridRef,
+        setGridRef,
+    ] = useState<MutableRefObject<TypeComputedProps | null> | null>(null);
+
+    const columns: any[] = [
+        {
+            name: 'timestamp',
+            header: 'Timestamp',
+            minWidth: 150,
+            defaultFlex: 1,
+            dateFormat: dateFormat,
+            filterEditor: DateFilter,
+            filterEditorProps: () => {
+                return { dateFormat: dateFormat };
+            },
+        },
     ];
 
+    const filters = [
+        { name: 'timestamp', operator: 'after', type: 'date', value: '' },
+    ];
+
+    for (const [id, data] of Object.entries(channelTemplates)) {
+        const template = data as ChannelTemplate;
+
+        //populate columns based on first entry
+        if (logger.channelTemplate == id) {
+            template.keys.forEach((key) => {
+                if (key != 'timestamp') {
+                    columns.push({ name: key, header: key, defaultFlex: 1 });
+                }
+            });
+        }
+    }
+
     let rows: any[] = [];
 
     logger.data.forEach((dataPoint, index) => {
-        rows.push({
-            id: index,
-            timestamp: dataPoint.timestamp,
-        });
+        var row = Object.assign({}, dataPoint);
+        row['id'] = index;
+        row['timestamp'] = moment(dataPoint.timestamp).format(dateFormat);
```

Comment:
Since the timestamp will always be there, it can be removed from the keys definition in the channel template so we don't need this line. If we want to render the timestamp differently it can be done with the render method in the column definition.

Reply:
I assume you mean `row['timestamp'] = moment(dataPoint.timestamp).format(dateFormat);`.  I'll remove it